src/logworker.cpp

```diff
@@ -112,14 +114,23 @@ namespace g3 {
 
    void LogWorker::addWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
       auto bg_addsink_call = [this, sink] {_impl._sinks.push_back(sink);};
-      auto token_done = g3::spawn_task(bg_addsink_call, _impl._bg.get());
-      token_done.wait();
+      g3::spawn_task(bg_addsink_call, _impl._bg.get());
```

Comment:
Maybe I commented on the commit instead?  I'm redoing the comment here. 

This code is the same as calling `token_done.wait()`. `spawn_task(...)` returns a `std::future` if a future isn't captured it falls out of scope and it means the same thing as calling `wait()`. 

So. In that regard I think you should use `send(...)` instead. similar to this:
![screen shot 2016-07-20 at 8 53 10 am](https://cloud.githubusercontent.com/assets/331742/16991483/f2c6d4ec-4e58-11e6-88d3-feb11b419efb.png)

As you pointed out. Adding a sink does not need to wait since any messages sent after this call will be in the queue and will be received by the new sink.